src/Util/String.cc

```diff
@@ -22,3 +22,11 @@ void SplitString(std::string& input, char delim, std::vector<std::string>& parts
         }
     }
 }
+
+bool EndsWith(const std::string& haystack, const std::string& needle) {
+    if (needle.size() > haystack.size()) {
+        return false;
+    }
+
+    return std::equal(needle.rbegin(), needle.rend(), haystack.rbegin());
+}
```

Comment:
I'd replace this with a case-insensitive regex check, and rename it to something more specific like `HasSuffix`, `DetectSuffix`, `SuffixFuzzy`, `SuffixIgnoreCase`, `EndsWithIgnoreCase`, etc.. Not sure how clear we need to be, but `EndsWith` would probably be misleading in a function that's in `Util`.